### PR TITLE
Move datastore code into a subpackage

### PR DESCRIPTION
**Pull request by dhermes**
This PR was created via: https://gist.github.com/dhermes/e239691aa584bd56a5352e34aad27cf3

```
export PROJECT_DIR="${HOME}/google-cloud-python"
export READMES_DIR="${HOME}/i-wrote-some-readmes-for-2357"
cd ${PROJECT_DIR}
git worktree add -b make-datastore-subpackage ../hotfix official/master
python make_commits.py \
  --git-root "${PROJECT_DIR}/../hotfix" \
  --package datastore \
  --package-name "Google Cloud Datastore" \
  --readme "${READMES_DIR}/datastore/README.rst" \
  --extra-requires $'    \'grpcio >= 1.0.0\',\n' \
  --maybe-omit $'omit =\n    */_generated/*.py\n'
```

**NOTE**: Since only `bigtable` and `datastore` had to touch the `run_pylint` and `tox.ini` configs for ignoring generated code, the 3rd commit is just written the old-fashioned way.

### CODE REVIEW

File: datastore/setup.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Topic :: Internet',
+    ],
+}
+
+
+REQUIREMENTS = [
+    'google-cloud-core',
+    'grpcio >= 1.0.0',
```
Comment (jonparrott) [inline datastore/setup.py]: No upper bound?

File: datastore/setup.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
+        'Topic :: Internet',
+    ],
+}
+
+
+REQUIREMENTS = [
+    'google-cloud-core',
```
Comment (jonparrott) [inline datastore/setup.py]: No bounds?

File: datastore/setup.py (hunk 1 above)
Comment (dhermes) [inline datastore/setup.py]: Tossed this in #2404 as something to resolve. SGTY?

File: datastore/setup.py (hunk 2 above)
Comment (dhermes) [inline datastore/setup.py]: (Ditto, in #2404)

File: datastore/setup.py (hunk 1 above)
Comment (jonparrott) [inline datastore/setup.py]: SGTM. 👍

File: .travis.yml [hunk 3]
```diff
@@ -7,11 +7,14 @@ install:
...
   - tox -e py27
   - (cd core && tox -e py27)
+  - (cd datastore && tox -e py27)
   - tox -e py34
   - (cd core && tox -e py34)
+  - (cd datastore && tox -e py34)
   - tox -e lint
   - tox -e cover
   - (cd core && tox -e cover)
+  - (cd datastore && tox -e cover)
```
Comment (daspecster) [inline .travis.yml]: I wonder what the build times will be once all the packages are split if we do this.

I know there is a desire, but do you have a plan on how to only build the modified packages?

**Comment by tseaver** [comment]
LGTM

**Comment by dhermes** [comment]
As with #2416, Pushed to resolve rebase conflict and re-ran script. Won't wait on Travis since a) that could be awhile and b) this will be painful.

**Pull request merged**